www/src/WorkerMapViewService.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+declare let self: Worker;
+import { GeoCoordinates } from "@here/harp-geoutils";
+import { MapView, RenderEvent } from "@here/harp-mapview";
+import { setAppBaseUrl } from "@here/harp-utils";
+import { EventEmitter } from "events";
+import { createMap } from "./map";
+
+export function init() {
+    let map!: MapView;
+    const eventHandler = new EventEmitter();
+
+    self.addEventListener("message", (event: MessageEvent) => {
+        const message = event.data;
+        switch (message.type) {
+            case "mapview-worker-init":
+                {
+                    setAppBaseUrl(message.baseUrl);
+                    const canvas: OffscreenCanvas = message.canvas;
```

Comment:
That's really cool. Would be awesome if you could make it an official harp.gl feature instead of implementing it in the webpage

Reply:
Thanks, glad someone likes this.

Frankly, i did it mostly to optimize "main thread" idle KPI of page speed index, but yeah.
The problem is this "offscreen" map is completely different beast that our main `MapView`.
You can't pass `OmvDataSource`, you can't register listener in efficient way, you can't access 3JS stuff inside.
The interface must be 100% async and plain-object based (if we want to keep it lean and mean).
We can can think about how to make good API from this in separate task.

And it means yet another worker bundle - and we love plaing with custom webpack configs.
And [it works only in Chrome](https://caniuse.com/#search=offscreencanvas) - ok not, only but key players like Safari/iOS and, Firefox don't even plan releasing it soon.